src/metabase/models/card.clj

```diff
@@ -655,12 +655,33 @@
       (pre-update-check-sandbox-constraints changes)
       (assert-valid-type (merge old-card-info changes)))))
 
+(defn- add-query-description-to-metric-card
+  "Add `:query_description` key to returned card.
+
+  Some users were missing description that was present in v1 metric API responses. This new key compensates for that.
+
+  This function is used in `t2/define-after-select :model/Card`. Metadata provider caching should be considered when
+  fetching multiple metric cards having common database, as done in eg. dashboard API context."
+  [card]
+  (if-not (and (map? card)
+               (= :metric (:type card))
+               (-> card :dataset_query not-empty)
+               (-> card :database_id))
+    card
+    (or (when-some [suggested-name (some-> (lib.metadata.jvm/application-database-metadata-provider
```

Comment:
nit: `m/assoc-some` ?

Reply:
That makes the expression way simpler. Thank you.